### PR TITLE
docs: updates from review of tuning content

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation updates**

Updates the [Tuning Kafka configuration](https://strimzi.io/docs/operators/in-development/configuring.html#assembly-tuning-config-str) content in the _Configuring guide_ following a review 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/managing/con-broker-config-properties.adoc [hunk 1]
```diff
@@ -109,10 +109,11 @@ As an exception, you might want to reduce the settings in a single-broker _test_
 Network threads handle requests to the Kafka cluster, such as produce and fetch requests from client applications.
 Produce requests are placed in a request queue. Responses are placed in a response queue.
 
-The number of network threads should reflect the replication factor and the levels of activity from client producers and consumers interacting with the Kafka cluster.
+The number of network threads per listener should reflect the replication factor and the levels of activity from client producers and consumers interacting with the Kafka cluster.
```
Comment (PaulRMellor) [inline documentation/modules/managing/con-broker-config-properties.adoc:112]: @mimaison -- do we we need to say something about prefixing with listener name?

File: documentation/assemblies/managing/assembly-tuning-config.adoc [hunk 2]
```diff
@@ -23,7 +23,7 @@ The following tools help with Kafka tuning:
 
 * xref:cruise-control-concepts-str[Cruise Control] generates optimization proposals that you can use to assess and implement a cluster rebalance
 * xref:proc-setting-broker-limits-str[Kafka Static Quota plugin] sets limits on brokers
-* xref:type-Rack-reference[Rack configuration] spreads broker partitions across racks
+* xref:type-Rack-reference[Rack configuration] spreads broker partitions across racks and allows consumers to fetch data from the nearest replica
```
Comment (scholzj) [inline documentation/assemblies/managing/assembly-tuning-config.adoc:26]: The Rack-awareness is just one of many things which need to be achieved to consume from nearest replica. So I think we need to be careful about this to make sure it does not suggest that it is all what is needed.

The distribution of the replicas across the partitions is also significantly more important for reliability and avauilability. So I think we need to make sure it does not get lost in this.

File: documentation/modules/managing/con-broker-config-properties.adoc (hunk 1 above)
Comment (scholzj) [reply documentation/modules/managing/con-broker-config-properties.adoc:112]: I don't think we currently allow to configure this differently for different listeners.

Comment (mimaison) [reply documentation/modules/managing/con-broker-config-properties.adoc:112]: It's new in 3.2 so yes we don't have it yet.

Comment (PaulRMellor) [reply documentation/modules/managing/con-broker-config-properties.adoc:112]: Okay, thanks

File: documentation/assemblies/managing/assembly-tuning-config.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/assemblies/managing/assembly-tuning-config.adoc:26]: > The Rack-awareness is just one of many things which need to be achieved to consume from nearest replica. So I think we need to be careful about this to make sure it does not suggest that it is all what is needed.

@scholzj Should we mention `affinity` rules and `topologySpreadConstraints` here? Or were you thinking of something else?

Comment (PaulRMellor) [reply documentation/assemblies/managing/assembly-tuning-config.adoc:26]: > The distribution of the replicas across the partitions is also significantly more important for reliability and avauilability. So I think we need to make sure it does not get lost in this.

We could add a note under these links:

_If you're considering rack configuration, make sure that topic  replication is set up for high availability. Having replicated topics across brokers, you can use rack awareness to distribute the replicas across racks._

Comment (scholzj) [reply documentation/assemblies/managing/assembly-tuning-config.adoc:26]: TBH, I would be fine to leave it as it was. The chapter it links to has the details. But if you want to change this, then it should be clear to focus on the main use-case for rack awareness which is availability and reliability. The fetching from closest replica is a niche use-case with a lot of gotchas and caveats.

Comment (scholzj) [reply documentation/assemblies/managing/assembly-tuning-config.adoc:26]: > We could add a note under these links:
> _If you're considering rack configuration, make sure that topic replication is set up for high availability. Having replicated topics across brokers, you can use rack awareness to spread partition replicas across racks._

Its not like this is wrong. It is more that I don't feel like it adds too much nd at the end you still need to go to https://strimzi.io/docs/operators/in-development/full/configuring.html#type-Rack-reference and read the whole stuff there.

Comment (PaulRMellor) [reply documentation/assemblies/managing/assembly-tuning-config.adoc:26]: Thanks. I'll leave it as it is.

**Review by scholzj — APPROVED** [review]

**Pull request merged**